src/main/java/servece/impl/ProcessorServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package servece.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.ProcessorService;
+import service.handler.OperationHandler;
+
+public class ProcessorServiceImpl implements ProcessorService {
+
+    @Override
+    public Map<String, Integer> processOnData(List<FruitTransaction> transactions,
+                                              Map<FruitTransaction.Operation,
+                                                      OperationHandler> operation) {
+        Map<String, Integer> fruitBalance = new HashMap<>();
+        for (FruitTransaction fruitTransaction : transactions) {
+            OperationHandler operationHandler =
+                    operation.get(fruitTransaction.getOperation());
+            if (operationHandler != null) {
+                operationHandler.apply(fruitTransaction, fruitBalance);
+            } else {
+                throw new RuntimeException("Operation does not find!");
+            }
```

Comment:
in cases like this, invert condition, throw early, remove else 
include operation in the message
rephrase to `couldn't be found`